src/openrct2/ride/Vehicle.cpp

```diff
@@ -2215,7 +2215,7 @@ static void train_ready_to_depart(rct_vehicle* vehicle, uint8_t num_peeps_on_tra
     vehicle->SetState(VEHICLE_STATUS_WAITING_FOR_PASSENGERS);
 }
 
-static int ride_get_train_index_from_vehicle(Ride* ride, uint16_t spriteIndex)
+static int32_t ride_get_train_index_from_vehicle(Ride* ride, uint16_t spriteIndex)
 {
     uint32_t trainIndex = 0;
```

Comment:
change this type to `int32_t` as well

Reply:
I had, actually, but when comparing it to `std::size(ride->num_vehicles)` it errored out due to narrowing. I thought about doing `static_cast` but was unsure. What do you think?

Reply:
What about refactoring the whole thing to use a std::optional? It would then get rid of the super edge cases.

Reply:
Will try this tonight, if it's too big

Reply:
I might have misunderstood, did you mean change the `ride->vehicles` to be an optional?
I think it only errored on the `else if (trainIndex >= std::size(ride->vehicles))`, not sure about the `if (trainIndex >= ride->num_vehicles)`. Sorry for the confusion.

Reply:
I was meaning for the return of the function. It gets rid of the invalid train index.